src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/ListingGenerationController.java

```diff
@@ -225,44 +202,96 @@
-        log.info("GENERATE HEARING DOCUMENT ---> " + LOG_MESSAGE + listingRequest.getCaseDetails().getCaseId());
+        log.info("GENERATE REPORT ---> " + LOG_MESSAGE + listingRequest.getCaseDetails().getCaseId());
 
         if (!verifyTokenService.verifyTokenSignature(userToken)) {
             log.error(INVALID_TOKEN, userToken);
             return ResponseEntity.status(FORBIDDEN.value()).build();
         }
 
+        var listingData = listingService.generateReportData(listingRequest.getCaseDetails(), userToken);
+
+        return getResponseEntity(listingData,listingRequest.getCaseDetails().getCaseTypeId(), userToken);
+
+    }
+
+    private ResponseEntity getResponseEntity(ListingData listingData, String caseTypeId, String userToken){
         List<String> errors = new ArrayList<>();
-        var listingData = listingRequest.getCaseDetails().getCaseData();
-        if ((listingData.getListingCollection() != null && !listingData.getListingCollection().isEmpty())
-                || ListingHelper.isReportType(listingData.getReportType())) {
-            var documentInfo = listingService.processHearingDocument(
-                    listingData, listingRequest.getCaseDetails().getCaseTypeId(), userToken);
-            listingData.setDocMarkUp(documentInfo.getMarkUp());
+
+        if(hasNonEmptyListings(listingData)){
+            var documentInfo = getDocumentInfo(listingData, caseTypeId, userToken);
+
+            updateListingDocMarkUp(listingData, documentInfo);
             return ResponseEntity.ok(ListingCallbackResponse.builder()
                     .data(listingData)
                     .significant_item(Helper.generateSignificantItem(documentInfo, errors))
                     .build());
-        } else {
-            errors.add("No hearings have been found for your search criteria");
+        } else{
             return ResponseEntity.ok(ListingCallbackResponse.builder()
-                    .errors(errors)
+                    .errors(setNoListingsErrorMessage(errors))
                     .data(listingData)
                     .build());
         }
     }
 
+    private void updateListingDocMarkUp(ListingData listingData, DocumentInfo documentInfo){
+        listingData.setDocMarkUp(documentInfo.getMarkUp());
+    }
+
+    private DocumentInfo getDocumentInfo(ListingData listingData, String caseTypeId, String userToken){
+        return listingService.processHearingDocument(listingData, caseTypeId, userToken);
+    }
+
+    private boolean hasNonEmptyListings(ListingData listingData) {
+
+        var isListingsCollNotEmpty = listingData.getListingCollection() != null &&
+                !listingData.getListingCollection().isEmpty();
+        var isAllowedReportType = ListingHelper.isReportType(listingData.getReportType());
+
+        return (isListingsCollNotEmpty || isAllowedReportType);
+    }
+
+    private List<String> setNoListingsErrorMessage(List<String> errors){
```

Comment:
You are returning the same object passed in as a parameter. This is unnecessary. How about
`private void addNoListingsErrorMessage(List<String> errors)`

Reply:
I see you point. I simplify it.